File: crates/jabuti-core/tests/fixtures/kotlin/units.kt
package demo

class Config(val depth: Int) {
    fun doubled(): Int {
        val scale = { value: Int -> value * 2 }
        return scale(depth)
    }
}

object Registry {
    fun register(name: String) {}
}

interface Visitor {
    fun visit(depth: Int): Boolean = depth > 0
}

fun outer(): Int {
    fun inner(): Int = 1
    return inner()
}

File: crates/jabuti-core/tests/fixtures/code/rust/units.rs
mod parser {
    pub struct Config {
        pub depth: usize,
    }

    impl Config {
        pub fn new(depth: usize) -> Self {
            Self { depth }
        }

        pub fn doubled(&self) -> usize {
            let scale = |value: usize| value * 2;
            scale(self.depth)
        }
    }

    pub trait Visitor {
        fn visit(&self, depth: usize) -> bool {
            depth > 0
        }
    }
}

enum Mode {
    Fast,
    Deep,
}

fn outer() -> usize {
    fn inner() -> usize {
        1
    }

    inner()
}
